File: app/src/main/java/cl/figonzal/evaluatool/dialogs/CorregidoDialogFragment.java
/*-----------------------------------------------------------------------------
                                                                              -
 This file is subject to the terms and conditions defined in                  -
 file 'LICENSE', which is part of this source code package                    -
                                                                              -
 Autor: Felipe González                                                       -
 Email: felipe.gonzalezalarcon94@gmail.com                                    -
                                                                              -
 Copyright (c) 2020                                                           -
                                                                              -
 Last modified 04-04-20 17:32                                                 -
 -----------------------------------------------------------------------------*/

package cl.figonzal.evaluatool.dialogs;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.google.android.material.button.MaterialButton;
import com.google.firebase.crashlytics.FirebaseCrashlytics;

import cl.figonzal.evaluatool.R;

public class CorregidoDialogFragment extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = requireActivity().getLayoutInflater();
        @SuppressLint("InflateParams") View view = inflater.inflate(R.layout.pd_corregido_dialog_layout, null);
        builder.setView(view);

        MaterialButton btn_entendido;

        btn_entendido = view.findViewById(R.id.btn_entendido_pd);
        btn_entendido.setOnClickListener(v -> {
            Log.d(getString(R.string.DIALOGO_AYUDA), getString(R.string.DIALOGO_AYUDA_MSG_CERRADO));

            FirebaseCrashlytics.getInstance().log(getString(R.string.DIALOGO_AYUDA) + getString(R.string.DIALOGO_AYUDA_MSG_CERRADO));

            dismiss();
        });

        return builder.create();
    }
}

File: app/src/main/java/cl/figonzal/evaluatool/dialogs/CorregidoDialogFragment.kt
/*

 This file is subject to the terms and conditions defined in
 file 'LICENSE', which is part of this source code package

 Autor: Felipe González
 Email: felipe.gonzalezalarcon94@gmail.com

 Copyright (c) 2021

 Last modified 18-04-21 22:17
 */
package cl.figonzal.evaluatool.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.view.LayoutInflater
import androidx.fragment.app.DialogFragment
import cl.figonzal.evaluatool.R
import cl.figonzal.evaluatool.databinding.PdCorregidoDialogLayoutBinding
import cl.figonzal.evaluatool.utilidades.logInfo

/**
 * Class that handle CorrectedDialogFragment for directPoints (PD) of a any student.
 * This is only a simple dialog fragment (Informative fragment with static information)
 *
 * @version 18-04-2021
 */
class CorregidoDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val builder = AlertDialog.Builder(activity)

        with(PdCorregidoDialogLayoutBinding.inflate(LayoutInflater.from(context)), {
            btnEntendidoPd.setOnClickListener {
                activity?.logInfo(R.string.DIALOGO_AYUDA_MSG_CERRADO)
                dismiss()
            }

            builder.setView(root)
        })
        return builder.create()
    }
}